core/src/test/java/brooklyn/location/access/PortForwardManagerRebindTest.java

```diff
@@ -86,6 +96,41 @@ public void testAssociationPreservedOnRebind() throws Exception {
     }
     
     @Test
+    public void testAssociationPreservedOnStateExport() throws Exception {
+        String publicIpId = "5.6.7.8";
+        String publicAddress = "5.6.7.8";
+
+        TestEntity origEntity = origApp.createAndManageChild(EntitySpec.create(TestEntity.class).impl(MyEntity.class));
+        PortForwardManager origPortForwardManager = origEntity.getConfig(MyEntity.PORT_FORWARD_MANAGER);
+
+        origPortForwardManager.associate(publicIpId, HostAndPort.fromParts(publicAddress, 40080), origSimulatedMachine, 80);
+
+        String label = origManagementContext.getManagementNodeId()+"-"+Time.makeDateSimpleStampString();
+        PersistenceObjectStore targetStore = BrooklynPersistenceUtils.newPersistenceObjectStore(origManagementContext, null, 
+            "tmp/web-persistence-"+label+"-"+Identifiers.makeRandomId(4));
+        File dir = ((FileBasedObjectStore)targetStore).getBaseDir();
+        // only register the parent dir because that will prevent leaks for the random ID
+        Os.deleteOnExitEmptyParentsUpTo(dir.getParentFile(), dir.getParentFile());
+        BrooklynPersistenceUtils.writeMemento(origManagementContext, targetStore, MementoCopyMode.LOCAL);            
+
+        RebindTestUtils.waitForPersisted(origApp);
+        log.info("Using manual export dir "+dir+" for rebind instead of "+mementoDir);
```

Comment:
I'm testing rest-like manual export distinct from normal persist-and-rebind.
On 18 Mar 2015 12:17, "Aled Sage" <notifications@github.com> wrote:

> In
> core/src/test/java/brooklyn/location/access/PortForwardManagerRebindTest.java
> <https://github.com/apache/incubator-brooklyn/pull/554#discussion_r26657630>
> :
>
> > +
> > +        TestEntity origEntity = origApp.createAndManageChild(EntitySpec.create(TestEntity.class).impl(MyEntity.class));
> > +        PortForwardManager origPortForwardManager = origEntity.getConfig(MyEntity.PORT_FORWARD_MANAGER);
> > +
> > +        origPortForwardManager.associate(publicIpId, HostAndPort.fromParts(publicAddress, 40080), origSimulatedMachine, 80);
> > +
> > +        String label = origManagementContext.getManagementNodeId()+"-"+Time.makeDateSimpleStampString();
> > +        PersistenceObjectStore targetStore = BrooklynPersistenceUtils.newPersistenceObjectStore(origManagementContext, null,
> > +            "tmp/web-persistence-"+label+"-"+Identifiers.makeRandomId(4));
> > +        File dir = ((FileBasedObjectStore)targetStore).getBaseDir();
> > +        // only register the parent dir because that will prevent leaks for the random ID
> > +        Os.deleteOnExitEmptyParentsUpTo(dir.getParentFile(), dir.getParentFile());
> > +        BrooklynPersistenceUtils.writeMemento(origManagementContext, targetStore, MementoCopyMode.LOCAL);
> > +
> > +        RebindTestUtils.waitForPersisted(origApp);
> > +        log.info("Using manual export dir "+dir+" for rebind instead of "+mementoDir);
>
> Why use the manual export dir? If we need to do that here, why do we not
> need to do it in all our other tests?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/apache/incubator-brooklyn/pull/554/files#r26657630>.
>